Add task timeout support. (#1317)
astronomer/astronomer-cosmos: docs/configuration/task-timeout.rst

.. _task-timeout:

Task Timeout
================

> In Airflow, the "execution_timeout" parameter allows you to set the maximum runtime for a Task.

Reviewer: ```suggestion
In Airflow, the ``execution_timeout`` parameter allows you to set the maximum runtime for a task.
```

> With Cosmos, you can specify an execution_timeout for each dbt model converted to a Task.
> This lets users set a threshold for the maximum runtime of a model, triggering a timeout error if the execution exceeds this limit.
> 
> By adding cosmos_task_timeout to the config of a dbt model, Cosmos will automatically apply the specified timeout to the Task based on this value.

Reviewer: ```suggestion
In Cosmos, you can apply an ``execution_timeout`` to each dbt model task by specifying a ``cosmos_task_timeout`` in the model’s configuration, which sets a runtime threshold to trigger a timeout error if exceeded.
```

Author: Thank you for the review; I’ve made the necessary corrections.
